src/gradients.rs

```diff
@@ -66,41 +68,97 @@ pub fn horizontal_sobel(image: &GrayImage) -> Image<Luma<i16>> {
     filter3x3(image, &HORIZONTAL_SOBEL)
 }
 
+#[cfg(feature = "rayon")]
+#[cfg_attr(docsrs, doc(cfg(feature = "rayon")))]
+/// Convolves an image with the [`HORIZONTAL_SOBEL`](static.HORIZONTAL_SOBEL.html)
+/// kernel to detect horizontal gradients. This version uses rayon to parallelize the computation.
+pub fn horizontal_sobel_parallel(image: &GrayImage) -> Image<Luma<i16>> {
+    filter3x3_parallel(image, &HORIZONTAL_SOBEL)
+}
+
```

Comment:
I'm not sure it's a good idea to create all these `_parallel` versions of the gradient functions as they are soon to be deprecated. Users are being encouraged to use `filter_clamped(image, kernel)` instead, which in this case they would simply change to `filter_clamped_parallel(image, kernel)`.

Reply:
Yep, sorry @sunsided - you've just run into an unfortunate merge conflict!

Reply:
Ah, sneaky! Just saw the `master` merge. I'll rebase and adjust. :)

Reply:
@ripytide @theotherphil This actually seems to have made things a lot easier. I brought in the current `master` and updated the implementations and tests.

I did attempt to update e.g. the `canny` functions as well but eventually reverted it; the measurement error on the benchmarks exceeded the average duration of the unparallelized version by far. So as of now, only `filter_clamped_parallel` in general and `sharpen3x3_parallel` as a special case it is.